### PR TITLE
docs: add bilingual Part II Chapter 2 concepts and code walkthrough

### PR DESCRIPTION
**Pull request by Ccyest**
Replace the Part II Chapter 2 placeholders with a concept lesson and a companion code walkthrough in both Chinese and English (four documents). The lessons follow a request through mini-sglang, explain process boundaries and their tradeoffs, and trace message types and source files. Both editions include request-flow and sampling diagrams, with the logits → softmax → sampling path and greedy branch explained explicitly.

Both editions follow the writing templates and are linked from the website sidebars. Chinese filenames use Chinese titles; companion coding documents end in `_代码.md` for Chinese and `_code.md` for English. Document these naming conventions and add CI checks with regression tests for title language and misplaced or missing coding suffixes.

Validation:
- `python3 -B -m unittest discover -s scripts -p 'test_check_chapters.py'` — 16 tests passed
- `python3 -B scripts/check_chapters.py --all` — 54 files, no errors
- `python3 -B scripts/check_site.py`
- `npm run build`
- `python3 -B scripts/check_site.py --dist docs/.vitepress/dist`

Inference commands were not rerun; this PR changes course content, navigation, and documentation checks.

### CODE REVIEW

File: course-material/ch/part2/第2章_code部分_一个请求的旅程.md [hunk 1]
```diff
@@ -0,0 +1,267 @@
...
+```python
+# engine/engine.py（简化）
+def forward_batch(self, batch, args):
+    logits = self.model.forward()                     # 前向：算出候选 token 的分数
+    next_tokens = self.sampler.sample(logits, args)   # 采样：从分布里挑一个 token
+    next_tokens_cpu = next_tokens.to("cpu", non_blocking=True)   # 结果拷回 CPU
+    ...
+```
+
+logits 可以理解为模型给词表中每个候选 token 打的分，还不是概率。greedy 直接选分数最高的；带温度的采样则先调整分数、转换成概率，再抽取一个 token。下一章会动手写前向和生成循环，到时再仔细看这两步。
```
Comment (1iyouzhen) [inline course-material/ch/part2/第2章_code部分_一个请求的旅程.md]: 这句话改成：

> 其中，logits 可以理解为模型给词表中每个候选 token 打的分还不是概率，经过 softmax 之后才会变成对应 token 的概率。而最终如何从这些概率里选出下一个 token，则取决于采样方法，比如：greedy 直接选分数最高的；带温度的采样则先调整分数、转换成概率，再按概率抽取一个 token。下一章会动手写前向和生成循环，到时再仔细看这两步。

会不会好理解一些嘞？

File: course-material/ch/part2/第2章_code部分_一个请求的旅程.md [hunk 2]
```diff
@@ -0,0 +1,267 @@
...
+    return forward_output
+```
+
+进入 `engine.forward_batch` 后，模型先算出 logits，采样器再选出下一个 token，最后把结果从 GPU 拷贝回 CPU，交给 scheduler 处理：
+
+```python
+# engine/engine.py（简化）
+def forward_batch(self, batch, args):
+    logits = self.model.forward()                     # 前向：算出候选 token 的分数
+    next_tokens = self.sampler.sample(logits, args)   # 采样：从分布里挑一个 token
```
Comment (1iyouzhen) [inline course-material/ch/part2/第2章_code部分_一个请求的旅程.md]: 这个 sample 前面要不要再加一个 softmax 过程，以免误解得到的 logits 可以直接用到 sample 阶段？

File: course-material/ch/part2/第2章_code部分_一个请求的旅程.md (hunk 1 above)
Comment (Ccyest) [reply course-material/ch/part2/第2章_code部分_一个请求的旅程.md]: 确实，有道理

File: course-material/ch/part2/第2章_code部分_一个请求的旅程.md (hunk 2 above)
Comment (Ccyest) [reply course-material/ch/part2/第2章_code部分_一个请求的旅程.md]: 嗯嗯，加了个图说明
